lib/cryptoexchange/exchanges/rightbtc/services/market.rb

```diff
@@ -0,0 +1,66 @@
+module Cryptoexchange::Exchanges
+  module Rightbtc
+    module Services
+      class Market < Cryptoexchange::Services::Market
+        class << self
+          def supports_individual_ticker_query?
+            false
+          end
+        end
+        TARGETS = ['ETH', 'BTC', 'BCY', 'ETP']
+
+        def fetch
+          output = super ticker_url
+          adapt_all(output)
+        end
+
+        def ticker_url
+          "#{Cryptoexchange::Exchanges::Rightbtc::Market::API_URL}/tickers"
+        end
+
+        def adapt_all(output)
+          markets = output['result']
+          markets.map do |market|
+            pair = market['market']
+            base, target = strip_pairs(pair)
+            market_pair = Cryptoexchange::Models::MarketPair.new(
+                base: base,
+                target: target,
+                market: Rightbtc::Market::NAME
+            )
+            adapt(market, market_pair)
+          end
+        end
+
+        def adapt(output, market_pair)
+          ticker = Cryptoexchange::Models::Ticker.new
+          ticker.base      = market_pair.base
+          ticker.target    = market_pair.target
+          ticker.market    = Rightbtc::Market::NAME
+          ticker.ask       = NumericHelper.to_d(output['sell'])
+          ticker.bid       = NumericHelper.to_d(output['buy'])
+          ticker.last      = NumericHelper.to_d(output['last24h'])
+          ticker.high      = NumericHelper.to_d(output['high24h'])
+          ticker.low       = NumericHelper.to_d(output['low24h'])
+          ticker.volume    = NumericHelper.to_d(output['vol24h'].abs)
+          ticker.timestamp = Time.now.to_i
+          ticker.payload   = output
+          ticker
+        end
+
+        def strip_pairs(pair)
+          #Match last 3 or 4 if it hits target
```

Comment:
@estherleongym What do you think of this approach?
Make additional call to `pairs` api to get the list of possible unique targets rather than defining it in a constant.
Extra call yes, but at least we are making only 2 calls and dont have to worry about the target currencies?

Reply:
@tmlee okay agreed; revised it to call the `pairs` api and to get a match from the `tickers` output.